tests/streaming-events-telemetry.md

```diff
@@ -0,0 +1,140 @@
+# Overview
+The goal is to test the events system from publishing end point to external receiving end point that tests every supported event which includes validation against the schema.
+
+# Scope
+This system is platform independent. This can be run on a fully configured & functioning SONiC system. As this involves BGP & Interface startup shutdown, require system with established BGP sessions.
+
+# Test bed
+Any testbed with couple of established BGP sessions.
+
+# Setup configuration
+Require: Few established BGP sessions and all *expected* dockers running.
+
+# Test
+This test will simulate scenarios to get the event fired and verify the fired event. It would also test cacheing, timeout, heartbeat & stats.
+
+# Test - Event receiver tool
+To test end to end, we need a tool that receives via gNMI
+
+1) Build a gNMI client tool (Go is easier) as part of SONiC-gNMI.
```

Comment:
sonic-gnmi has provided gnmi_get, gnmi_set and gnmi_cli

Reply:
You are correct. But we can't use gnmi-get or set for on change. The gnmi-cli would work, but it does not have an option to save response only in a o/p file. So a custom tool, that can provide the JSON strings in o/p file will be helpful to parse the o/p for test purposes. Additionally we could customize this tool with options as needed for testing, which may not be required in a user facing gnmi-cli tool.
Moreover the tool's code is from open-config and hence any update has to be via patch only, which adds additional restrictions.
Writing this tool should be pretty straight forward. We could use openconfig code as reference.